sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/Jsr305AnnotationsCheck.java

```diff
@@ -0,0 +1,996 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2019 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.github.sevntu.checkstyle.checks.coding;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import antlr.collections.AST;
+import com.github.sevntu.checkstyle.SevntuUtil;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.FullIdent;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * Jsr305AnnotationsCheck - a check to enforce use of nullness anotations.
+ * The standalone version of this plugin resides under
+ * <a href="https://github.com/bjrke/JSR305CheckstylePlugin">
+ * https://github.com/mbert/JSR305CheckstylePlugin</a>
+ * <p>
+ * Copyright (C) 2008 Marcus Thiesen (initial version) Copyright (C) 2008-2019 Jan Burkhardt
+ * (maintainer)
+ * <br>
+ * Thanks to Mattias Nissler, Thorsten Ehlers, Fabian Loewner, Ole Langbehn for contributions.
+ * </p>
+ * <p>
+ * This check is activated by setting the package(s) to check:
```

Comment:
Unfortunately javadoc on Check is the only source of documentation for Checks in sevntu, please copy almost all from issue description to javadoc, users need to know all details of you Check and example of work to be absolutely clear.

Reply:
OK, done. Here's the result: 

![screenshot-2019-04-18-11:00:55](https://user-images.githubusercontent.com/383271/56349676-50987400-61c9-11e9-9b9f-af3dc65e531f.png)

Reply:
please add few(1-2) examples (with inlined comments of where is violation) to Javadoc, user do like examples more documentation.
example - https://checkstyle.org/config_annotation.html#AnnotationLocation_Examples
https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/annotation/AnnotationLocationCheck.java#L91

Reply:
Good point. Done.